apps/frontend/src/pages/quiz/libs/components/quiz-form/quiz-form.tsx

```diff
@@ -59,14 +60,44 @@ const QuizForm: React.FC<Properties> = ({ onNext }: Properties) => {
 		void dispatch(quizActions.previousQuestion());
 	}, [dispatch]);
 
-	const handleNextStep = useCallback(() => {
-		if (isLast) {
-			return;
-		}
+	const checkIfValueIsUndefined = useCallback(
+		(formData: FormToSave, keysToCheck: string[]): boolean => {
+			return keysToCheck.some((key) => formData[key] === undefined);
+		},
+		[],
+	);
+
+	const getAnswerIds = useCallback((formData: FormToSave) => {
+		return Object.values(formData).map(Number);
+	}, []);
+
+	const handleNextStep = useCallback(
+		(data: QuizFormValues) => {
+			if (category) {
+				const ids = category.map(
+					(categoryItem) => `answer${categoryItem.id.toString()}`,
+				);
+				const hasUndefined = checkIfValueIsUndefined(data, ids);
+
+				if (!hasUndefined) {
+					const newObject = Object.fromEntries(
+						Object.entries(data).filter(([key]) => key !== "answers"),
```

Comment:
Since we don't need `answers`, let's remove it totally

Reply:
this `answers` is actually from default value